hydra-configs-torchvision/tests/test_instantiate_datasets.py

```diff
@@ -0,0 +1,106 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+import os
+import shutil
+import pytest
+from pathlib import Path
+from hydra.utils import get_class, instantiate
+from omegaconf import OmegaConf
+
+import torchvision.datasets as datasets
+
+from typing import Any
+
+
+@pytest.mark.parametrize(
+    "modulepath, classname, cfg, passthrough_args, passthrough_kwargs, expected_class",
+    [
+        pytest.param(
+            "datasets.vision",
+            "VisionDataset",
+            {"root": None},
+            [],
+            {},
+            datasets.VisionDataset,
+            id="VisionDatasetConf",
+        ),
+        pytest.param(
+            "datasets.mnist",
+            "MNIST",
+            {"root": None},
+            [],
+            {},
+            datasets.MNIST,
+            id="MNISTConf",
+        ),
+        pytest.param(
+            "datasets.mnist",
+            "FashionMNIST",
+            {"root": None},
+            [],
+            {},
+            datasets.FashionMNIST,
+            id="FashionMNISTConf",
+        ),
+        pytest.param(
+            "datasets.mnist",
+            "KMNIST",
+            {"root": None},
+            [],
+            {},
+            datasets.KMNIST,
+            id="KMNISTConf",
+        ),
+        # TODO: These tests will need to be changed after blockers:
+        # 1. EMNISTConf and QMNISTConf are manually created
+        # 2. hydra.utils.instantiate is updated to allow *kwargs instantiation
+        #        pytest.param(
```

Comment:
Assuming those tests are the ones that need the test data, can we have custom tests for them that actually loads (and maybe even cache) said data?
I don't think it makes sense to make the repo 200MB.

Reply:
I agree. I think I made an attempt to do this, but something about it didn't work out. I'll revisit now.

Reply:
Its done.

Reply:
what did you do?

Reply:
I figured out the correct way to create a dummy dataset dynamically within the tmpdir pytest fixture. So instead of needing to use a `.pt` binary to load a dataset, I just create one. It's literally a float tensor with 1 element, but it's all that's necessary to test that a dataset class can be instantiated correctly.